### PR TITLE
#2292-me, fixing mobile double click for table rows.

### PR DESCRIPTION
**Pull request by ionut-movila**
This should fix the double click on mobile, but still due to missing pagination it will be working very slow (not really usable).

I have used a new lib to handle mobile double tap: https://github.com/JedWatson/react-hammerjs, because with a simple test when running on mobile view in browser `onDoubleClick` will not fire, see http://fiddle.jshell.net/jxg/z2hdqw7L/show/

### CODE REVIEW

**Comment by teosarca** [comment]
task: https://github.com/metasfresh/metasfresh-webui-frontend/issues/2292

File: src/components/table/Table.js [hunk 1]
```diff
@@ -1301,6 +1305,13 @@ class Table extends Component {
             handleToggleExpand={toggleFullScreen}
           />
         )}
+        {isMobileOrTablet && rows.length > MOBILE_TABLE_SIZE_LIMIT && (
+          <span className="text-danger">
+            {/*TODO: use here translations, but how we define new keys?*/}
```
Comment (siemiatj) [inline src/components/table/Table.js]: @ionut-movila ping @teosarca or @metas-ts and ask them to add whatever translations you need

File: src/components/table/Table.js [hunk 2]
```diff
@@ -967,93 +969,95 @@ class Table extends Component {
...
-          return collapsedRows.indexOf(row[keyProperty]) === -1;
+    let renderRows = rows.filter(row => {
+      if (collapsedRows) {
+        return collapsedRows.indexOf(row[keyProperty]) === -1;
+      }
+      return true;
+    });
+
+    if (isMobileOrTablet && rows.length > MOBILE_TABLE_SIZE_LIMIT) {
+      renderRows = renderRows.slice(0, MOBILE_TABLE_SIZE_LIMIT);
```
Comment (siemiatj) [inline src/components/table/Table.js]: So how do you switch to the next page if pagination is hidden on mobile devices ?

Comment (teosarca) [reply src/components/table/Table.js]: my suggestion, as a quick&dirty solution, was to limit the query to a reasonable number of rows.
If that amount is hit, we display the "limit" message at the bottom of the page.
User has to filter and search if he needs more.
In this way we make the window usable with small amount of effort.

----
THE solution: live scroll or some pagination which looks nice on mobile... but I think live scroll is THE shit!

File: src/components/table/Table.js (hunk 1 above)
Comment (teosarca) [reply src/components/table/Table.js]: introduced `view.limitTo`.
See https://w101.metasfresh.com:8443/rest/api/i18n/messages?filter=view.limitTo for current value.

File: src/components/table/Table.js (hunk 2 above)
Comment (siemiatj) [reply src/components/table/Table.js]: Yeah, we have an issue to implement that. The only problem is when we're showing nested data.

**Comment by siemiatj** [comment]
Just use the new translation and it's good to merge.

**Comment by teosarca** [comment]
task https://github.com/metasfresh/metasfresh-webui-frontend/issues/2149

**Comment by teosarca** [comment]
@ionut-movila  still have issues with this one. Let's check it together on monday.

File: src/components/table/Table.js (hunk 1 above)
Comment (ionut-movila) [reply src/components/table/Table.js]: I have fixed this with commit: https://github.com/metasfresh/metasfresh-webui-frontend/pull/2290/commits/ccd7caaf1b0d7470acb1c6a816fadd4ec411196f

@metas-ts But we need to update the string from backend with respect to interpolation rules of the FE lib: https://www.npmjs.com/package/counterpart#interpolation

So it should be like this: "Limited to %(limit)s rows of %(total)s"

Comment (teosarca) [reply src/components/table/Table.js]: done

**Comment by teosarca** [comment]
quickly checked but i got:
image

**Pull request merged**